modules/cudaarithm/src/cuda/transpose.cu

```diff
@@ -60,34 +60,147 @@ void cv::cuda::transpose(InputArray _src, OutputArray _dst, Stream& stream)
 {
     GpuMat src = getInputMat(_src, stream);
 
+    const int srcType = src.type();
+    const int srcDepth = src.depth();
+    const int srcCn = src.channels();
     const size_t elemSize = src.elemSize();
+    const size_t elemSize1 = src.elemSize1();
 
-    CV_Assert( elemSize == 1 || elemSize == 4 || elemSize == 8 );
+    //CV_Assert( elemSize == 1 || elemSize == 4 || elemSize == 8 );
 
     GpuMat dst = getOutputMat(_dst, src.cols, src.rows, src.type(), stream);
 
-    if (elemSize == 1)
+    const bool isNppiNativelySupported =
+      (srcType == CV_8UC1)  || (srcType == CV_8UC3)  || (srcType == CV_8UC4)  ||
+      (srcType == CV_16UC1) || (srcType == CV_16UC3) || (srcType == CV_16UC4) ||
+      (srcType == CV_16SC1) || (srcType == CV_16SC3) || (srcType == CV_16SC4) ||
+      (srcType == CV_32SC1) || (srcType == CV_32SC3) || (srcType == CV_32SC4) ||
+      (srcType == CV_32FC1) || (srcType == CV_32FC3) || (srcType == CV_32FC4);
+    const bool isElemSizeSupportedByNppi =
+      ((elemSize != 0) && !(elemSize%1) && ((elemSize/1)<=4)) ||
+      ((elemSize != 0) && !(elemSize%2) && ((elemSize/2)<=4)) ||
+      ((elemSize != 0) && !(elemSize%4) && ((elemSize/4)<=4)) ||
+      ((elemSize != 0) && !(elemSize%8) && ((elemSize/8)<=2));
+    if (src.empty())
+      dst.release();
```

Comment:
Empty input should be a error as nobody want to process "nothing" in real use cases: https://github.com/opencv/opencv/issues/8300

Reply:
Ok, I thought it was better since [Allow empty matrices in most functions](https://github.com/opencv/opencv/wiki/2021#2021-11-17) (personnaly I prefer a no-op rather than an exception)

Reply:
I will also add a fastpath for size == (1, 1), where transpose is just a copy